iceoryx_posh/test/integrationtests/test_publisher_subscriber_communication.cpp

```diff
@@ -126,6 +157,111 @@ TEST_F(PublisherSubscriberCommunication_test, AllSubscriberInterfacesCanBeSubscr
+{
+    ::testing::Test::RecordProperty("TEST_ID", "31cbd36d-32f1-4bc7-9980-0cdf5f248035");
+
+    constexpr uint64_t historyRequest = 1;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_FALSE(publisher->hasSubscribers());
+}
+
+TEST_F(PublisherSubscriberCommunication_test, SubscriberRequiringNoHistoryDoesConnectToPublisherWithNoHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "c47f5ebd-044c-480b-a4bb-d700655105ac");
+
+    constexpr uint64_t historyRequest = 1;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = false;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_TRUE(publisher->hasSubscribers());
+}
+
+TEST_F(PublisherSubscriberCommunication_test,
+       SubscriberRequiringHistoryDoesConnectToPublisherWithSufficientHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "0ca391fe-c4f6-48b5-bd36-96854513c6bb");
+
+    constexpr uint64_t historyRequest = 3;
+    constexpr uint64_t historyCapacity = 3;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_TRUE(publisher->hasSubscribers());
+}
+
+TEST_F(PublisherSubscriberCommunication_test,
+       SubscriberRequiringHistoryDoesNotConnectToPublisherWithInSufficientHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "46b917e6-75f1-4cd2-8ffa-1c254f3423a7");
+
+    constexpr uint64_t historyRequest = 3;
+    constexpr uint64_t historyCapacity = 2;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_FALSE(publisher->hasSubscribers());
+}
+
+TEST_F(PublisherSubscriberCommunication_test, SubscriberRequiringHistoryDoesNotConnectToPublisherWithNoHistorySupport)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "46b917e6-75f1-4cd2-8ffa-1c254f3423a7");
+
+    constexpr uint64_t historyRequest = 3;
+    constexpr uint64_t historyCapacity = 0;
+    constexpr bool requireHistory = true;
+
+    auto publisher = createPublisher<int>(historyCapacity);
+    auto subscriber = createSubscriber<int>(historyRequest, requireHistory);
+
+    this->InterOpWait();
+
+    ASSERT_TRUE(publisher);
+    EXPECT_FALSE(publisher->hasSubscribers());
+}
```

Comment:
Déjà vu ... isn't this already tested on line 160?

Reply:
Yes, basically. Case is redundant (3 or 1 should not matter and if it does we could argue for other numbers as well).